parallel-install/pkg/deployment/overrideinterceptors_test.go

```diff
@@ -224,71 +281,299 @@ func Test_DomainNameOverrideInterceptor(t *testing.T) {
+
+	gardenerCM := fakeGardenerCM()
+
+	t.Run("test default cert for local cluster", func(t *testing.T) {
+		// given
+		kubeClient := fake.NewSimpleClientset()
+		interceptor := NewCertificateOverrideInterceptor("global.tlsCrt", "global.tlsKey", kubeClient)
+		interceptor.isLocalCluster = isLocalClusterFunc(true)
+
 		ob := OverridesBuilder{}
-		ob.AddInterceptor([]string{"global.tlsCrt", "global.tlsKey"}, NewCertificateOverrideInterceptor("global.tlsCrt", "global.tlsKey"))
-		// verify cert overrides
+
+		ob.AddInterceptor([]string{"global.tlsCrt", "global.tlsKey"}, interceptor)
+
+		// when
 		overrides, err := ob.Build()
+
+		// then
+		require.NoError(t, err)
+		require.NotEmpty(t, overrides.Map())
+		require.Equal(t, getOverride(overrides.Map(), "global.tlsCrt"), defaultLocalTLSCrtEnc)
+		require.Equal(t, getOverride(overrides.Map(), "global.tlsKey"), defaultLocalTLSKeyEnc)
+	})
+
+	t.Run("test default cert for remote non-gardener cluster", func(t *testing.T) {
+		// given
+		kubeClient := fake.NewSimpleClientset()
+		interceptor := NewCertificateOverrideInterceptor("global.tlsCrt", "global.tlsKey", kubeClient)
+		interceptor.isLocalCluster = isLocalClusterFunc(false)
+
+		ob := OverridesBuilder{}
+
+		ob.AddInterceptor([]string{"global.tlsCrt", "global.tlsKey"}, interceptor)
+
+		// when
+		overrides, err := ob.Build()
+
+		// then
+		require.NoError(t, err)
 		require.NotEmpty(t, overrides.Map())
+		require.Equal(t, getOverride(overrides.Map(), "global.tlsCrt"), defaultRemoteTLSCrtEnc)
+		require.Equal(t, getOverride(overrides.Map(), "global.tlsKey"), defaultRemoteTLSKeyEnc)
+	})
+
+	t.Run("test default cert is not set for a gardener cluster", func(t *testing.T) {
+		kubeClient := fake.NewSimpleClientset(gardenerCM)
+
+		// given
+		interceptor := NewCertificateOverrideInterceptor("global.tlsCrt", "global.tlsKey", kubeClient)
+		interceptor.isLocalCluster = isLocalClusterFunc(true) //Try to trick the code into local domain. Should be handled properly
+
+		ob := OverridesBuilder{}
+
+		ob.AddInterceptor([]string{"global.tlsCrt", "global.tlsKey"}, interceptor)
+
+		// when
+		overrides, err := ob.Build()
+
+		// then
 		require.NoError(t, err)
+		require.Empty(t, overrides.Map()) //No override injected
```

Comment:
The assertion is pretty straight-forward. Do we need the comment here? Or maye move it to the line starting with `// then..`?

Reply:
Removed